doc/reference/networking/lwm2m.rst

```diff
@@ -11,8 +11,281 @@ Overview
+`Device` object with some data and the callback we defined above:
+
+.. code-block:: c
+
+   /*
+    * Server URL of default Security object = 0/0/0
+    * Use leshan.eclipse.org server IP (5.39.83.206) for connection
+    */
+   lwm2m_engine_set_string("0/0/0", "coap://5.39.83.206");
+
+   /*
+    * Security Mode of default Security object = 0/0/2
+    * 3 = NoSec mode (no security beware!)
+    */
+   lwm2m_engine_set_u8("0/0/2", 3);
+
+   #define CLIENT_MANUFACTURER "Zephyr Manufacturer"
+
+   /*
+    * Manufacturer resource of Device object = 3/0/0
+    * We use lwm2m_engine_set_res_data() function to set a pointer to the
+    * CLIENT_MANUFACTURER string.
+    * Note the LWM2M_RES_DATA_FLAG_RO flag which stops the engine from
+    * trying to assign a new value to the buffer.
+    */
+   lwm2m_engine_set_res_data("3/0/0", CLIENT_MANUFACTURER,
+                             sizeof(CLIENT_MANUFACTURER),
+                             LWM2M_RES_DATA_FLAG_RO);
+
+   /* Reboot resource of Device object = 3/0/4 */
+   lwm2m_engine_register_exec_callback("3/0/4", device_reboot_cb);
+
+Lastly, we start the LwM2M RD client (which in turn starts the LwM2M engine).
+The second parameter of `lwm2m_rd_client_start` is the client endpoint name.
+This is important as it needs to be unique per LwM2M server:
+
+.. code-block:: c
+
+   (void)memset(&client, 0x0, sizeof(client));
+   lwm2m_rd_client_start(&client, "unique-endpoint-name", rd_client_event);
+
+Using LwM2M library with DTLS
+*****************************
+
+The Zephyr LwM2M library can be used with DTLS transport for secure
+communication by selecting `CONFIG_LWM2M_DTLS_SUPPORT`.  In the client
+initialization we need to create a PSK and identity.  These need to match
+the security information loaded onto the LwM2M server.  Normally, the
+endpoint name is used to lookup the related security information:
+
+.. code-block:: c
+
+   /* "000102030405060708090a0b0c0d0e0f" */
+   static unsigned char client_psk[] = {
+       0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07,
+       0x08, 0x09, 0x0a, 0x0b, 0x0c, 0x0d, 0x0e, 0x0f
+   };
+   static const char client_identity[] = "Client_identity";
+
+Next we alter the `Security` object resources to include DTLS security
+information.  The server URL should begin with `coaps://` to indicate security
+is required.  Assign a 0 value (Pre-shared Key mode) to the `Security Mode`
+resource.  Lastly, set the client identity and PSK resources.
+
+.. code-block:: c
+
+   /* Use coaps:// for server URL protocol */
+   lwm2m_engine_set_string("0/0/0", "coaps://5.39.83.206");
+   /* 0 = Pre-Shared Key mode */
+   lwm2m_engine_set_u8("0/0/2", 0);
+   /* Set the client identity */
+   lwm2m_engine_set_string("0/0/3", (char *)client_identity);
+   /* Set the client pre-shared key (PSK) */
+   lwm2m_engine_set_opaque("0/0/5", (void *)client_psk, sizeof(client_psk));
+
+Before calling lwm2m_rd_client_start() assign the tls_tag # where the LwM2M
```

Comment:
You can add refenreces to functions too like this
```
:c:func:`lwm2m_rd_client_start()`
```
Please apply to other functions mentioned in this doc.